dissect/cstruct/types/structure.py

```diff
@@ -401,6 +407,46 @@ def __repr__(self) -> str:
 
         return f"<{self.__class__.__name__} {' '.join(values)}>"
 
+    @property
+    def __sizes__(self) -> Mapping[str, int | None]:
+        return ChainMap(self.__class__.__static_sizes__, self.__dynamic_sizes__)
+
+    @property
+    def __values__(self) -> MutableMapping[str, Any]:
+        return StructureValuesProxy(self)
+
+
+class StructureValuesProxy(MutableMapping):
+    """A proxy for the values of fields of a Structure."""
+
+    def __init__(self, struct: Structure):
+        self._struct: Structure = struct
+        # Unfortunately, we need to support code that adds attributes dynamically :/
+        self._fields: set[str] = set(iter(self._struct.__class__.fields))
```

Comment:
Alternatively, we could use a ChainMap here over field names and extra fields. This would be more memory efficient, but slightly slower when iterating (since chainmap needs to preserve key order).

Reply:
What do you mean extra fields? You can't dynamically add fields to an already parsed structure.

Reply:
See above